src/otx/core/data/dataset/classification.py

```diff
@@ -89,3 +93,171 @@ def _convert_to_onehot(self, labels: torch.tensor) -> torch.tensor:
+        """
+
+        def get_exclusive_group_info(all_groups: list[Label | list[Label]]) -> dict[str, Any]:
+            """Get exclusive group information."""
+            exclusive_groups = [g for g in all_groups if len(g) > 1]
+            exclusive_groups.sort(key=itemgetter(0))
+
+            last_logits_pos = 0
+            num_single_label_classes = 0
+            head_idx_to_logits_range = {}
+            class_to_idx = {}
+
+            for i, group in enumerate(exclusive_groups):
+                head_idx_to_logits_range[str(i)] = (last_logits_pos, last_logits_pos + len(group))
+                last_logits_pos += len(group)
+                for j, c in enumerate(group):
+                    class_to_idx[c] = (i, j)
+                    num_single_label_classes += 1
+
+            return {
+                "num_multiclass_heads": len(exclusive_groups),
+                "exclusive_groups": exclusive_groups,
+                "head_idx_to_logits_range": head_idx_to_logits_range,
+                "class_to_idx": class_to_idx,
+                "num_single_label_classes": num_single_label_classes,
+            }
+
+        def get_single_label_group_info(
+            all_groups: list[Label | list[Label]],
+            num_exclusive_groups: int,
+        ) -> dict[str, Any]:
+            """Get single label group information."""
+            single_label_groups = [g for g in all_groups if len(g) == 1]
+            single_label_groups.sort(key=itemgetter(0))
+
+            class_to_idx = {}
+
+            for i, group in enumerate(single_label_groups):
+                class_to_idx[group[0]] = (num_exclusive_groups, i)
+
+            return {
+                "num_multilabel_classes": len(single_label_groups),
+                "class_to_idx": class_to_idx,
+            }
+
+        def merge_class_to_idx(
+            exclusive_ctoi: dict[str, tuple[int, int]],
+            single_label_ctoi: dict[str, tuple[int, int]],
+        ) -> dict[str, tuple[int, int]]:
+            """Merge the class_to_idx information from exclusive and single_label groups."""
+
+            def put_key_values(src: dict, dst: dict) -> None:
+                """Put key and values from src to dst."""
+                for k, v in src.items():
+                    dst[k] = v
+
+            class_to_idx: dict[str, tuple[int, int]] = {}
+            put_key_values(exclusive_ctoi, class_to_idx)
+            put_key_values(single_label_ctoi, class_to_idx)
+            return class_to_idx
+
+        all_groups = [label_group.labels for label_group in self.dm_categories.label_groups]
+
+        exclusive_group_info = get_exclusive_group_info(all_groups)
+        single_label_group_info = get_single_label_group_info(all_groups, exclusive_group_info["num_multiclass_heads"])
+
+        merged_class_to_idx = merge_class_to_idx(
+            exclusive_group_info["class_to_idx"],
+            single_label_group_info["class_to_idx"],
+        )
+
+        return HLabelInfo(
```

Comment:
Can we wrap this function into
```suggestion
        return HLabelInfo.from_dm_label_groups(self.dm_categories.label_groups)
```

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2761/commits/fdc033ec6b1fdc6615705ddd75af960a49af9f1d